### ISSUE TITLE
Log suppress feature is not honoring the log_suppress_interval setting

### ISSUE THREAD

**Issue by lecaros** [fluent/fluent-bit#7051]
## Bug Report

**Describe the bug**
The log suppress feature is not honoring the `log_suppress_interval` setting. Instead, it uses the `cache->timeout`, which is a fixed value of 10 (see description below).

The suspicion was also reported [here](https://github.com/fluent/fluent-bit/pull/6435#issuecomment-1371621189)
 

**To Reproduce**

Run Fluent Bit with the following config.

fluent-bit.conf

```text
[SERVICE]
    flush 1
[INPUT]
    name dummy

[OUTPUT]
    Name         http
    Match        *
    host        localhost
    port        1234
    log_suppress_interval 15
    workers 1
```

The `workers 1` is intended to reduce the verbosity. The log suppression feature seems to be per worker.

**Expected behavior**
It should honor the setting in the `log_suppress_interval` setting.

**Screenshots**

**Your Environment**
* Version used: master branch
* Configuration: provided above
* Environment name and version (e.g. Kubernetes? What version?):
* Server type and version:
* Operating System and version:
* Filters and plugins:

**Additional context**

The repro above will show these error messages (other lines suppressed for clarity):

```text
[2023/03/22 17:06:58] [error] [output:http:http.0] no upstream connections available to localhost:1234
...
[2023/03/22 17:07:08] [error] [output:http:http.0] no upstream connections available to localhost:1234
```

Those lines are 10 seconds apart instead of 15, as configured in the config file. This is where the check is done:

https://github.com/fluent/fluent-bit/blob/2226fd6c524529f9d2774fa53dde6bc74c86f5c3/src/flb_log.c#L295

That line compares timestamp + cache timeout with the current timestamp. However, the cache timeout is a fixed value of 10, as seen in the code below:

https://github.com/fluent/fluent-bit/blob/2226fd6c524529f9d2774fa53dde6bc74c86f5c3/src/flb_log.c#L330

This is why it always repeats the message after 10 seconds instead of the value in `log_suppress_interval`.

The fix should be to set this cache timeout to the value of `log_suppress_interval`, unless other changes are required, like making this setting part of the log worker context or the flb_log structure.

**Comment by edsiper** [comment]
cc: @nokute78

**Comment by nokute78** [comment]
It is easy to add `log_suppress_interval` to `flb_config` and we can pass it to flb_log_cache_create as a system wide value.
However it is not enough to set different values for each plugin.

Should we support `log_suppress_interval` for each plugins ?

**Comment by lecaros** [comment]
Yes. The feature should cover plugins and the other components (service, network, engine).
Currently, it's a parameter added to output plugins, and there's your patch to add it to input and filters.

**Comment by nokute78** [comment]
> Yes. The feature should cover plugins and the other components (service, network, engine).

Current design can't handle them.
Each `flb_worker`s have a `log_cache` and timeout which is 10.
It means log_suppress_interval feature is for worker not for components.

In addition, this feature works using `flb_plg_*` APIs. 
If we use `flb_error/info/log ...`, this feature doesn't work. 

If we want to support intervals for components, we need a log queue for each components
We also need a new logging API to indicate the component.  e.g. `flb_new_error(component, message...)`
As you know, we need to replace current logging APIs.
It will need a huge refactoring.

**Comment by lecaros** [comment]
The refactoring involving the other components could be made later.
Can that cache timeout be defined by `log_suppress_interval`? This parameter is not being used now, hence not producing the expected behavior.

**Comment by nokute78** [comment]
We support two logging API
- System API e.g. `flb_error`
- Plugin API e.g. `flb_plg_error`

In current design, the system API doesn't check suppress setting.
Because each plugin instances have an interval and it is not passed for the system API.
It means some of logs  (of flb_plg_*) are suppressed and others ( of flb_** ) are not. It is the reason that refactoring is needed.

We may be able to set more rough system wide setting.

**Comment by dh185221** [comment]
should this be reopened?

**Comment by lecaros** [comment]
ping @edsiper @nokute78

**Comment by drbugfinder-work** [comment]
We experience the same behavior with the log suppress option. 
Every 10 seconds, still one message gets printed: 
`[2024/01/11 09:12:17] [ warn] [output:loki:loki.1] empty record accessor key translation for pattern: $optional_label`

Looks like this is the culprit as @lecaros already showed: 
https://github.com/fluent/fluent-bit/blob/2226fd6c524529f9d2774fa53dde6bc74c86f5c3/src/flb_log.c#L330

**Comment by asymmetric** [comment]
So to understand, does the `log_suppress_interval` option have no effect at all, or is the problem that it does not work in 100% of the cases?
